pulp_ansible/app/tasks/collections.py

```diff
@@ -430,10 +430,32 @@ def _update_highest_version(collection_version):
     equals False on the last highest version and True on this version.
     Otherwise does nothing.
     """
+
     last_highest = collection_version.collection.versions.filter(is_highest=True).first()
     if not last_highest:
-        collection_version.is_highest = True
+        # we only have one version, so mark it as the highest
+        if collection_version.collection.versions.count() == 1:
+            collection_version.is_highest = True
+            collection_version.save()
+            return None
+
+        # compute highest from the whole list ...
+        highest = None
+        for cv in collection_version.collection.versions.all():
+            sv = Version(cv.version)
+            if highest is None or (highest[0] < sv and not sv.prerelease):
+                highest = (sv, cv)
```

Comment:
I think this is wrong when all the versions are pre-releases. The first pre-release will be set to `highest`, but none of the following ones will be properly checked even if they are higher.

Sidenote, I hope this portion of the method won't get called often (hopefully never), as it could be quite a performance hit during a sync if we called this part for each CV.

Reply:
Yep, you were right. I fixed it with a helper function to make the comparison and added a unit test for the scenario.